---
title: "Week 10 Questions"

format:
  html:
    code-fold: false
    code-tools: true
    code-copy: true
    highlight-style: github
    code-overflow: wrap
---

1.  **Looking back on the quarter what are two things you were not expecting to learn but did, glad you did, and how would you implement this knowledge beyond this class?**

  - a. Even though I never did any RNA-seq bench work, it was nice to know how the workflow went so that I could write the methods as part of my research project. If I ever do a project like this in the future, I will know how to do the enitre protocol from RNA extraction to Gene annotations. 
  
  - b. I really liked using Rpubs! It makes the code look really nice and prints out all the graphs for you so that the rendering is perfect. Much better way to share research than Github .Rmd files. 

2.  **Rank the following from 1-5, with 1 being your best friend and 5 being your enemy (raven, rstudio, knitr, bash, github)**

 - raven: 3
 - rstudio: 5
 - knitr: 4
 - bash: 4
 - github: 4

3.  **As part of your research project you must of learned something beyond methods. What is the most important biological finding from your work to to date?** Assuming you follow this line of research what is your long term goal of what you expect to learn about biology, ecology, or the environment?
  - best finding: gonad genes in the pacific geoduck are highly specialized and small relative to other tissue libraries such as the ctenidia. Also, that the 19 chromosome genome is highly conserved in Mollusks. 
  - long term goal: the goal of this project was to identify potential candidate genes for future gene knock-out experiments on oysters or clams. It is expected that this work will be a part of the future work, and that doing gene editing in non-model organisms is really hard! 
