---
title: '3.2-DEseq2'
author: "Olivia"
date: "2023-05-11"
output: 
  html_document:
    theme: readable
    highlight: zenburn
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
---

# Generate a plot and table of differentially expressed genes
## For this assignment you will be taking RNA-seq reads off the sequencer, and determining what genes are expressed higher in treatment group A compared to treatments group B. Why would someone want to do this? This can tell you something about the physiological response to a “treatment”, which generally speaking could be anything from environment, disease, developmental stage, tissue, species…

### create directory called "programs"
#ran this in terminal
olivia@raven:~/gitrepos/olivia-classwork$ pwd
/home/olivia/gitrepos/olivia-classwork
olivia@raven:~/gitrepos/olivia-classwork$ mkdir programs


### WGet kallisto 
#ran in terminal
olivia@raven:~/gitrepos/olivia-classwork$ cd programs
olivia@raven:~/gitrepos/olivia-classwork/programs$  wget https://github.com/pachterlab/kallisto/releases/download/v0.46.1/kallisto_linux-v0.46.1.tar.gz
#kallisto_linux-v0.46.1.tar.gz’ saved [7274037/7274037]
olivia@raven:~/gitrepos/olivia-classwork/programs$ tar -xvzf kallisto_linux-v0.46.1.tar.gz



### Uncompress the file
#ran in terminal
olivia@raven:~/gitrepos/olivia-classwork/programs$ cd kallisto
olivia@raven:~/gitrepos/olivia-classwork/programs/kallisto$ ls
kallisto  license.txt  README.md  test
olivia@raven:~/gitrepos/olivia-classwork/programs/kallisto$ ./kallisto
kallisto 0.46.1

Usage: kallisto <CMD> [arguments] ..

Where <CMD> can be one of:

    index         Builds a kallisto index 
    quant         Runs the quantification algorithm 
    bus           Generate BUS files for single-cell data 
    pseudo        Runs the pseudoalignment step 
    merge         Merges several batch runs 
    h5dump        Converts HDF5-formatted results to plaintext
    inspect       Inspects and gives information about an index
    version       Prints version information
    cite          Prints citation information

Running kallisto <CMD> without arguments prints usage information for <CMD>

# Running Kallisto 
## download reference genome
```{bash, eval=FALSE}
cd data
curl --insecure -O https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/rna.fna
```

#### make index file and rename it as cgigas_roslin_rna.index
```{bash, eval=FALSE}
/home/shared/kallisto/kallisto \
 index -i \
~/gitrepos/olivia-classwork/data/cgigas_roslin_rna.index \
~/gitrepos/olivia-classwork/data/rna.fna
```

#### download sequence reads
```{bash, eval=FALSE}
#ran in terminal
~/gitrepos/olivia-classwork/data
wget --recursive --no-parent --no-directories \
--no-check-certificate \
--accept '*.fastq.gz' \
https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/nopp/
```

### look for all files ending in L001_R1_001.fastq.gz
```{bash, eval=FALSE}
find ~/gitrepos/olivia-classwork/assignments/code/*fastq.gz \
| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ~/gitrepos/olivia-classwork/data/cgigas_roslin_rna.index \
-o ~/gitrepos/olivia-classwork/output/kallisto_01/{} \
-t 40 \
--single -l 100 -s 10 ~/gitrepos/olivia-classwork/assignments/code/{}_L001_R1_001.fastq.gz
```

```{bash, eval=FALSE}
#run in terminal
perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl \
--est_method kallisto \
    --gene_trans_map none \
    --out_prefix ~/gitrepos/olivia-classwork/output/kallisto_01 \
    --name_sample_by_basedir \
    ~/gitrepos/olivia-classwork/output/kallisto_01/D54_S145/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/D56_S136/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/D58_S144/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M45_S140/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M48_S137/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M89_S138/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/D55_S146/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/D57_S143/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/D59_S142/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M46_S141/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M49_S139/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/M90_S147/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N48_S194/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N50_S187/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N52_S184/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N54_S193/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N56_S192/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N58_S195/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N49_S185/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N51_S186/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N53_S188/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N55_S190/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N57_S191/abundance.tsv \
     ~/gitrepos/olivia-classwork/output/kallisto_01/N59_S189/abundance.tsv

```
# Running DESeq2
This code performs differential expression analysis to identify differentially expressed genes (DEGs) between a control condition and a desiccated condition.

First, it reads in a count matrix of isoform counts generated by Kallisto, with row names set to the gene/transcript IDs and the first column removed. It then rounds the counts to whole numbers.

Next, it creates a data.frame containing information about the experimental conditions and sets row names to match the column names in the count matrix. It uses this information to create a DESeqDataSet object, which is then passed to the DESeq() function to fit a negative binomial model and estimate dispersions. The results() function is used to extract the results table, which is ordered by gene/transcript ID.

The code then prints the top few rows of the results table and calculates the number of DEGs with an adjusted p-value less than or equal to 0.05. It plots the log2 fold changes versus the mean normalized counts for all genes, highlighting significant DEGs in red and adding horizontal lines at 2-fold upregulation and downregulation. Finally, it writes the list of significant DEGs to a file called “DEGlist.tab”.

### load packages
```{r}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.16") 
BiocManager::install("DESeq2")
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

## Read count matrix
```{r, eval=TRUE}
countmatrix <- read.delim("~/gitrepos/olivia-classwork/output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

## round up 
```{r}
countmatrix <- round(countmatrix, 0)
str(countmatrix)
```

# Get DEGs based on treatment (desiccation, spelled wrong in code)
```{r}
deseq2.colData <- data.frame(condition=factor(c(rep("control", 12), rep("desicated", 12))), 
                             type=factor(rep("single-read", 24)))
rownames(deseq2.colData) <- colnames(data)
deseq2.dds <- DESeqDataSetFromMatrix(countData = countmatrix,
                                     colData = deseq2.colData, 
                                     design = ~ condition)
```

```{r}
deseq2.dds <- DESeq(deseq2.dds)
deseq2.res <- results(deseq2.dds)
deseq2.res <- deseq2.res[order(rownames(deseq2.res)), ]
```

```{r}
head(deseq2.res)
```

## Count number of hits with adjusted p-value less then 0.05
```{r}
dim(deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ])
```

# the Main Plot
```{r}
tmp <- deseq2.res
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-3, 3), log="x", col="darkgray",
     main="DEG Dessication  (pval <= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```


```{r}
write.table(tmp.sig, "~/gitrepos/olivia-classwork/output/DEGlist.tab", row.names = T)
head(tmp.sig)
```

