---
title: "06-alignments-RNAseq"
author: "Olivia Cattau"
date: '2023-05-02'
output: html_document
---
# Introduction
This week, we will consider a different way of looking at the relatedness of genetic information: single nucleotide polymorphisms (SNPs). SNPs are used to describe population variation or phylogenetic relationships. A first step in this is aligning sequence reads to look for variation.


## SAM and BAM files
SAM (Sequence Alignment/Map) and BAM (Binary Alignment/Map) are two file formats used to store aligned sequencing data. Both formats are designed to store read alignments from high-throughput sequencing experiments to a reference genome. 

## SAMtools
Samtools is a software package designed for manipulating and analyzing high-throughput sequencing data, specifically SAM (Sequence Alignment/Map) and BAM (Binary Alignment/Map) files. 

# TASK 1
### look at alignment data

# Download alignmnet data
```{r, engine='bash', eval=FALSE}         
cd ../data
curl -O https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/120321-cvBS/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam
curl -O https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/120321-cvBS/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam.bai
```

```{r, engine='bash', eval=FALSE}         
cd ../data
curl -O https://gannet.fish.washington.edu/seashell/bu-mox/data/Cvirg-genome/GCF_002022765.2_C_virginica-3.0_genomic.fa
curl -O https://gannet.fish.washington.edu/seashell/bu-mox/data/Cvirg-genome/GCF_002022765.2_C_virginica-3.0_genomic.fa.fai
```

# View TVIEW
### Run in command line
/home/shared/samtools-1.12/samtools tview \
/home/olivia/gitrepos/olivia-classwork/assignments/code/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam \
/home/olivia/gitrepos/olivia-classwork/assignments/code/GCF_002022765.2_C_virginica-3.0_genomic.fa

see ![TVIEW_image](https://raw.githubusercontent.com/course-fish546-2023/olivia-classwork/main/assignments/output/assign-06/Screen%20Shot%202023-05-04%20at%2010.45.32%20AM.png)

# Task 2
## Aligning WGS data and visualizing in IGV

```{r, engine='bash', eval=FALSE}
cd ~/gitrepos/olivia-classwork/assignments/code #not ideal but what is done is done
curl -O https://owl.fish.washington.edu/nightingales/C_gigas/F143n08_R2_001.fastq.gz
curl -O https://owl.fish.washington.edu/nightingales/C_gigas/F143n08_R1_001.fastq.gz
```

```{r, engine='bash', eval=FALSE}
cd ~/gitrepos/olivia-classwork/data
curl -O https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa
curl -O https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa.fai
curl -O https://gannet.fish.washington.edu/panopea/Cg-roslin/GCF_902806645.1_cgigas_uk_roslin_v1_genomic-mito.gtf
```


# Alignment
```{r, engine='bash', eval=FALSE}
/home/shared/hisat2-2.2.1/hisat2-build \
-f /home/olivia/gitrepos/olivia-classwork/assignments/code/cgigas_uk_roslin_v1_genomic-mito.fa \
/home/olivia/gitrepos/olivia-classwork/assignments/output/cgigas_uk_roslin_v1_genomic-mito.index
```

```{r, engine='bash'}
/home/shared/hisat2-2.2.1/hisat2 \
-x /home/olivia/gitrepos/olivia-classwork/assignments/output/cgigas_uk_roslin_v1_genomic-mito.index \
-p 4 \
-1 /home/olivia/gitrepos/olivia-classwork/assignments/code/F143n08_R1_001.fastq.gz \
-2 /home/olivia/gitrepos/olivia-classwork/assignments/code/F143n08_R2_001.fastq.gz \
-S /home/olivia/gitrepos/olivia-classwork/assignments/output/F143_cgigas.sam
```

### start here, push now

## take a look
```{r, engine='bash'}
tail -1 ../output/F143_cgigas.sam
```

## Convert SAM to BAM, using 4 additional threads
```{r, engine='bash'}
/home/shared/samtools-1.12/samtools view -@ 4 -bS \
../output/F143_cgigas.sam > ../output/F143_cgigas.bam
```

## Sort the BAM file, using 4 additional thread
```{r, engine='bash'}
/home/shared/samtools-1.12/samtools sort -@ 4 \
../output/F143_cgigas.bam -o ../output/F143_cgigas_sorted.bam
```

## Index the sorted BAM file (multi-threading is not applicable to this operation)
```{r, engine='bash'}
/home/shared/samtools-1.12/samtools index \
../output/F143_cgigas_sorted.bam
```

# mpileup
### Now bcftools is recommended for mpileup instead of samtools (which was described in textbook)


```{r, engine='bash'}
/home/shared/bcftools-1.14/bcftools mpileup --threads 4 --no-BAQ \
--fasta-ref ../data/cgigas_uk_roslin_v1_genomic-mito.fa \
../output/F143_cgigas_sorted.bam > ../output/F143_mpileup_output.txt
```

```{r, engine='bash'}
tail ../output/F143_mpileup_output.txt
```

```{r, engine='bash'}
cat ../output/F143_mpileup_output.txt \
| /home/shared/bcftools-1.14/bcftools call -mv -Oz \
> ../output/F143_mpile.vcf.gz
```


```{r, engine='bash'}
zgrep "^##" -v ../output/F143_mpile.vcf.gz | \
awk 'BEGIN{OFS="\t"} {split($8, a, ";"); print $1,$2,$4,$5,$6,a[1],$9,$10}' | head

```


The code below might not work. That is fine. The VCF in the above chunk can be used for visualization in IGV.

```{r, engine='bash'}
/home/shared/bcftools-1.14/bcftools call \
-v -c ../output/F143_mpile.vcf.gz \
> ../output/F143_mpile_calls.vcf
```

# Visualize
- show bam file
- show (2) genome feature files
- bonus for annotating screenshots

